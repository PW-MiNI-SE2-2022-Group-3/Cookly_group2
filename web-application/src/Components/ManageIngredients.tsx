import React from "react";
import { useState, useEffect } from "react";

import Table from "@mui/material/Table";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import Button from "@mui/material/Button";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import Paper from "@mui/material/Paper";
import Box from "@mui/material/Box";
import TextField from "@mui/material/TextField";
import LinearProgress from "@material-ui/core/LinearProgress";

import Dialog from "@mui/material/Dialog";
import DialogTitle from "@mui/material/DialogTitle";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";

import TablePagination from "@mui/material/TablePagination";
import TableBody from "@mui/material/TableBody";
import TableFooter from "@mui/material/TableFooter";
import TablePaginationActions from "./TablePagination";

import axios from "axios";

import DeleteIcon from "@mui/icons-material/Delete";
import SearchIcon from "@mui/icons-material/Search";
import EditIcon from "@mui/icons-material/Edit";
import EggIcon from "@mui/icons-material/Egg";

import "../styles/Manage.css";
import { PATH } from "../Constants/API";

interface ManageIngredientsProps {}

const columns = ["NO.", "NAME", "ACTIONS"];

const ManageIngredients: React.FC<ManageIngredientsProps> = (
  props: ManageIngredientsProps
) => {
  type IngredientResponse = {
    id?: number;
    name?: string;
  };

  const [loading, setLoading] = useState(false);

  const [addIngredient, setAddIngredient] = useState(false);
  const [newIngredient, setNewIngredient] = useState<IngredientResponse>({});

  const [editIngredient, setEditIngredient] = useState(false);
  const [editIngredientData, setEditIngredientData] =
    useState<IngredientResponse>({});

  const [searchValue, setSearchValue] = useState("");

  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(5);

  const [data, setData] = useState<IngredientResponse[]>([]);

  useEffect(() => {
    const fetchData = async () => {
      getData();
    };

    fetchData();
  }, []);

  const getData = () => {
    axios
      .post(PATH + "/ingredients/all?page=0&limit=5000", {
        headers: {
          "Content-Type": "application/json",
          Authorization: "---",
        },
      })
      .then((response: { data: { ingredients: any; }; }) => {
        setData(response.data.ingredients);
      })
      .catch((err: any) => {
        alert(err);
      });
  };

  //Add Ingredient
  const addIngredientHandler = (event: any) => {
    setLoading(true);
    event.preventDefault();

    axios
      .post(PATH + "/ingredients", {
        name: newIngredient.name,
      })
      .then((response: any) => {
        setAddIngredient(false);
        setNewIngredient({});
        setLoading(false);
        getData();
      })
      .catch((err: any) => {
        setLoading(false);
        alert(err);
      });
  };

  //Edit Ingredient
  const editIngredientHandler = (event: any) => {
    setLoading(true);
    event.preventDefault();
    axios
      .put(
        PATH + "/ingredients?id=" + editIngredientData.id,
        editIngredientData,
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: "---",
          },
        }
      )
      .then((response: any) => {
        setEditIngredient(false);
        setEditIngredientData({});
        setLoading(false);
        getData();
      })
      .catch((err: { message: any; }) => {
        setLoading(false);
        alert(err.message);
      });
  };

  //Delete Ingredient
  const deleteButtonHandler = (event: any, ingredientId: number) => {
    axios
      .delete(PATH + "/ingredients/" + ingredientId, {
        headers: {
          "Content-Type": "application/json",
          Authorization: "---",
        },
      })
      .then((response: any) => {
        getData();
      })
      .catch((err: any) => {
        alert(err);
      });
  };

  //Filter Ingredients
  const filterIngredients = () => {
    axios
      .post(
        PATH + "/ingredients/all?page=0&limit=5000",
        { name: searchValue },
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: "---",
          },
        }
      )
      .then((response: { data: { ingredients: any; }; }) => {
        setData(response.data.ingredients);
      })
      .catch((err: any) => {
        alert(err);
      });
  };

  const handleChangePage = (event: any, newPage: number) => {
    setPage(newPage);
  };
  const emptyRows = () => {
    return page > 0 ? Math.max(0, (1 + page) * rowsPerPage - data.length) : 0;
  };
  const handleChangeRowsPerPage = (event: any) => {
    setPage(0);
    setRowsPerPage(parseInt(event.target.value, 10));
  };

  return data !== undefined ? (
    <Box style={{ width: "100%", margin: "auto", paddingTop: "20px" }}>
      <TextField
        id="outlined-basic"
        variant="standard"
        placeholder="Search"
        sx={{
          backgroundColor: "white",
          width: "30%",
          padding: "0px",
          marginBottom: "10px",
        }}
        InputProps={{
          startAdornment: <SearchIcon />,
        }}
        onChange={(event: { target: { value: string; }; }) => {
          setSearchValue(event.target.value);
          if (event.target.value !== "") filterIngredients();
          else getData();
        }}
      />
      <Button
        variant="contained"
        sx={{
          backgroundColor: "#c4560c",
          color: "white",
          float: "right",
          marginBottom: "10px",
          borderRadius: 0,
          width: "15%",
          "&:hover": { backgroundColor: "#d97938" },
        }}
        onClick={() => {
          setAddIngredient(true);
        }}
      >
        Add Ingredient
        <EggIcon style={{ marginLeft: "10px" }} />
      </Button>

      {/* data table */}
      <TableContainer
        component={Paper}
        className="tablecontainer"
        style={{ borderRadius: 0 }}
      >
        <Table size="small" className="table">
          <TableHead>
            <TableRow>
              {columns.map((name) => (
                <TableCell
                  key={name}
                  className="tablecellheader"
                  style={{
                    textAlign: name === "ACTIONS" ? "center" : "left",
                    color: "white",
                  }}
                >
                  {name}
                </TableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {(rowsPerPage > 0
              ? data.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
              : data
            ).map((d: any) => (
              <TableRow hover>
                <TableCell className="tablecell"> {d.id} </TableCell>
                <TableCell className="tablecell">{d.name}</TableCell>

                <TableCell
                  className="tablecell"
                  style={{ textAlign: "center", width: "35%" }}
                >
                  <Button
                    variant="contained"
                    style={{
                      backgroundColor: "darkred",
                      color: "white",
                      borderRadius: 0,
                      width: "200px",
                      margin: "5px",
                    }}
                    onClick={(event: any) => deleteButtonHandler(event, d.id)}
                    tabIndex={d.id}
                  >
                    Delete
                    <DeleteIcon style={{ marginLeft: "10px" }} />
                  </Button>
                  <Button
                    variant="contained"
                    style={{
                      backgroundColor: "green",
                      color: "white",
                      borderRadius: 0,
                      width: "200px",
                      margin: "5px",
                    }}
                    onClick={(event: any) => {
                      setEditIngredient(true);
                      setEditIngredientData({ name: d.name, id: d.id });
                    }}
                    tabIndex={d.id}
                  >
                    Edit
                    <EditIcon style={{ marginLeft: "10px" }} />
                  </Button>
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
          <TableFooter>
            <TableRow>
              <TablePagination
                rowsPerPageOptions={[5, 10, 25, { label: "All", value: -1 }]}
                colSpan={columns.length}
                count={data.length}
                rowsPerPage={rowsPerPage}
                page={page}
                SelectProps={{
                  inputProps: {
                    "aria-label": "rows per page",
                  },
                  native: true,
                }}
                onPageChange={handleChangePage}
                onRowsPerPageChange={handleChangeRowsPerPage}
                ActionsComponent={TablePaginationActions}
              />
            </TableRow>
          </TableFooter>
        </Table>
      </TableContainer>

      {/* add ingredient */}
      <Dialog
        sx={{
          "& .MuiDialog-paper": {
            width: "80%",
            maxHeight: 435,
            borderRadius: 0,
          },
        }}
        maxWidth="xs"
        open={addIngredient}
      >
        <DialogTitle
          sx={{
            backgroundColor: "#c4560c",
            color: "white",
          }}
        >
          ADD NEW INGREDIENT
        </DialogTitle>
        {loading && <LinearProgress />}
        <form
          onSubmit={(event: any) => {
            addIngredientHandler(event);
          }}
        >
          <DialogContent sx={{ marginTop: "20px" }}>
            <TextField
              autoFocus
              required
              margin="dense"
              id="name"
              label="Ingredient Name"
              type="text"
              fullWidth
              variant="standard"
              onChange={(event: { target: { value: any; }; }) => {
                setNewIngredient({ name: event.target.value });
              }}
            />
          </DialogContent>
          <DialogActions>
            <Button
              type="reset"
              color="error"
              onClick={() => {
                setAddIngredient(false);
              }}
            >
              Cancel
            </Button>
            <Button type="submit">Add</Button>
          </DialogActions>
        </form>
      </Dialog>

      {/* edit ingredient */}
      <Dialog
        sx={{
          "& .MuiDialog-paper": {
            width: "80%",
            maxHeight: 435,
            borderRadius: 0,
          },
        }}
        maxWidth="xs"
        open={editIngredient}
      >
        <DialogTitle
          sx={{
            backgroundColor: "#c4560c",
            color: "white",
          }}
        >
          EDIT INGREDIENT
        </DialogTitle>
        {loading && <LinearProgress />}
        <form
          onSubmit={(event: any) => {
            editIngredientHandler(event);
          }}
        >
          <DialogContent sx={{ marginTop: "20px" }}>
            <TextField
              autoFocus
              required
              margin="dense"
              value={editIngredientData.name}
              id="name"
              label="Ingredient Name"
              type="text"
              fullWidth
              variant="standard"
              onChange={(event: { target: { value: any; }; }) => {
                setEditIngredientData((prevState: any) => ({
                  ...prevState,
                  name: event.target.value,
                }));
              }}
            />
          </DialogContent>
          <DialogActions>
            <Button
              type="reset"
              color="error"
              onClick={() => {
                setEditIngredient(false);
              }}
            >
              Cancel
            </Button>
            <Button type="submit">Ok</Button>
          </DialogActions>
        </form>
      </Dialog>
    </Box>
  ) : (
    <></>
  );
};
export default ManageIngredients;
