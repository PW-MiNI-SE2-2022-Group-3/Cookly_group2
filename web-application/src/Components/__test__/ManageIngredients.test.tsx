import React, { useState } from "react";
import ReactDOM from "react-dom";
import {
  render,
  screen,
  fireEvent,
  queryByTestId,
  cleanup,
} from "@testing-library/react";
import ManageIngredients from "../ManageIngredients";

it("renders correctly without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ManageIngredients />, div);
});

it("renders buttons correctly", () => {
  const { getByTestId } = render(<ManageIngredients />);
  expect(getByTestId("add-button")).toBeInTheDocument();
  expect(getByTestId("add-button")).toHaveTextContent("Add Ingredient");
});

it("renders input fields correctly", () => {
  const { getByTestId } = render(<ManageIngredients />);
  expect(getByTestId("search-textfield")).toBeInTheDocument();
});

it("matches snapshot", () => {
  const tree = render(<ManageIngredients />);
  expect(tree).toMatchSnapshot();
});
