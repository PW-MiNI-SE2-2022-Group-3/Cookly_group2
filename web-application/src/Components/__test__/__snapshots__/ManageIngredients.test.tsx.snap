// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
        style="width: 100%; margin: auto; padding-top: 20px;"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-6u1dxy-MuiFormControl-root-MuiTextField-root"
          data-testid="search-textfield"
        >
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <input
              aria-invalid="false"
              class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="outlined-basic"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1t69hsd-MuiButtonBase-root-MuiButton-root"
          data-testid="add-button"
          tabindex="0"
          type="button"
        >
          Add Ingredient
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EggIcon"
            focusable="false"
            style="margin-left: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 3C8.5 3 5 9.33 5 14c0 3.87 3.13 7 7 7s7-3.13 7-7c0-4.67-3.5-11-7-11zm1 15c-3 0-5-1.99-5-5 0-.55.45-1 1-1s1 .45 1 1c0 2.92 2.42 3 3 3 .55 0 1 .45 1 1s-.45 1-1 1z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root tablecontainer css-11xur9t-MuiPaper-root-MuiTableContainer-root"
          style="border-radius: 0;"
        >
          <table
            class="MuiTable-root table css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: left; color: white;"
                >
                  NO.
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: left; color: white;"
                >
                  NAME
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                  scope="col"
                  style="text-align: center; color: white;"
                >
                  ACTIONS
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
            <tfoot
              class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-footer css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeSmall MuiTablePagination-root css-m0bqz3-MuiTableCell-root-MuiTablePagination-root"
                  colspan="3"
                >
                  <div
                    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
                  >
                    <div
                      class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                    />
                    <p
                      class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                      id="mui-14"
                    >
                      Rows per page:
                    </p>
                    <div
                      class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                      variant="standard"
                    >
                      <select
                        aria-label="rows per page"
                        class="MuiTablePagination-select MuiNativeSelect-select MuiNativeSelect-standard MuiInputBase-input css-m3l1ig-MuiNativeSelect-select-MuiInputBase-input"
                        id="mui-13"
                      >
                        <option
                          class="MuiTablePagination-menuItem"
                          value="5"
                        >
                          5
                        </option>
                        <option
                          class="MuiTablePagination-menuItem"
                          value="10"
                        >
                          10
                        </option>
                        <option
                          class="MuiTablePagination-menuItem"
                          value="25"
                        >
                          25
                        </option>
                        <option
                          class="MuiTablePagination-menuItem"
                          value="-1"
                        >
                          All
                        </option>
                      </select>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiNativeSelect-icon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                    </div>
                    <p
                      class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                    >
                      0â€“0 of 0
                    </p>
                    <div
                      class="MuiBox-root css-1zye22"
                    >
                      <button
                        aria-label="first page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="FirstPageIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="previous page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="KeyboardArrowLeftIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="next page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="KeyboardArrowRightIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                          />
                        </svg>
                      </button>
                      <button
                        aria-label="last page"
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="LastPageIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </td>
              </tr>
            </tfoot>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
      style="width: 100%; margin: auto; padding-top: 20px;"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-6u1dxy-MuiFormControl-root-MuiTextField-root"
        data-testid="search-textfield"
      >
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1ptx2yq-MuiInputBase-root-MuiInput-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="outlined-basic"
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1t69hsd-MuiButtonBase-root-MuiButton-root"
        data-testid="add-button"
        tabindex="0"
        type="button"
      >
        Add Ingredient
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EggIcon"
          focusable="false"
          style="margin-left: 10px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3C8.5 3 5 9.33 5 14c0 3.87 3.13 7 7 7s7-3.13 7-7c0-4.67-3.5-11-7-11zm1 15c-3 0-5-1.99-5-5 0-.55.45-1 1-1s1 .45 1 1c0 2.92 2.42 3 3 3 .55 0 1 .45 1 1s-.45 1-1 1z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root tablecontainer css-11xur9t-MuiPaper-root-MuiTableContainer-root"
        style="border-radius: 0;"
      >
        <table
          class="MuiTable-root table css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: left; color: white;"
              >
                NO.
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: left; color: white;"
              >
                NAME
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall tablecellheader css-1howxi1-MuiTableCell-root"
                scope="col"
                style="text-align: center; color: white;"
              >
                ACTIONS
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          />
          <tfoot
            class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-footer css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeSmall MuiTablePagination-root css-m0bqz3-MuiTableCell-root-MuiTablePagination-root"
                colspan="3"
              >
                <div
                  class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
                >
                  <div
                    class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                  />
                  <p
                    class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                    id="mui-14"
                  >
                    Rows per page:
                  </p>
                  <div
                    class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                    variant="standard"
                  >
                    <select
                      aria-label="rows per page"
                      class="MuiTablePagination-select MuiNativeSelect-select MuiNativeSelect-standard MuiInputBase-input css-m3l1ig-MuiNativeSelect-select-MuiInputBase-input"
                      id="mui-13"
                    >
                      <option
                        class="MuiTablePagination-menuItem"
                        value="5"
                      >
                        5
                      </option>
                      <option
                        class="MuiTablePagination-menuItem"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="MuiTablePagination-menuItem"
                        value="25"
                      >
                        25
                      </option>
                      <option
                        class="MuiTablePagination-menuItem"
                        value="-1"
                      >
                        All
                      </option>
                    </select>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiNativeSelect-icon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                  <p
                    class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                  >
                    0â€“0 of 0
                  </p>
                  <div
                    class="MuiBox-root css-1zye22"
                  >
                    <button
                      aria-label="first page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="FirstPageIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="previous page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="next page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="last page"
                      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LastPageIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
